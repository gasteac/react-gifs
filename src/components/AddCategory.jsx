import { useState } from "react";
import PropTypes from 'prop-types'

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("");
  const onInputChange = (e) => {
    setInputValue(e.target.value);
  };
  const onSubmit = (e) => {
    e.preventDefault();
    const newInputValue = inputValue.trim().toUpperCase();
    if (newInputValue.length <= 1) return;
    onNewCategory(newInputValue);
    setInputValue("");
  };
  return (
    <form onSubmit={onSubmit}>
      <input
        type="text"
        placeholder="Buscar Gifs"
        value={inputValue}
        onChange={onInputChange}
      />
      <button className="button-search" type="submit">Buscar</button>
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired
}