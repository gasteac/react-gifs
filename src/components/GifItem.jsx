import PropTypes from "prop-types";
export const GifItem = ({ title, url }) => {
  return (
    <div className="card w-96 h-96 bg-base-100 shadow-xl m-5">
      <figure>
        <img src={url} alt="gif" />
      </figure>
      <div className="card-body">
        <h2 className="card-title !text-black line-clamp-1">{title}</h2>
      </div>
    </div>
  );
};

GifItem.propTypes = {
  title : PropTypes.string.isRequired,
  url   : PropTypes.string.isRequired,
}

  // <div className="">
  //   <img src={url} alt={title} />
  //   <p>{title}</p>
  // </div>