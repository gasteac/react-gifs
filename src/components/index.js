//Solo para ralizar exportaciones mas simples

export * from "./AddCategory"
export * from "./GifGrid"
export * from "./GifItem"
