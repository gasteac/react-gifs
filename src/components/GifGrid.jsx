import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";


//Es la categoria en su, con su boton de eliminar, y sus gifs
export const GifGrid = ({category, handleDelete}) => {
  const {images, isLoading} = useFetchGifs(category)
  return (
    <>
      <div className="flex px-16 mb-6 gap-4 items-center">
        <h1 className="text-2xl font-semibold line-clamp-1 bg-gray-800 p-2 rounded-md w-full">{category}</h1>
        <button
          className="btn  bg-red-500 text-white"
          onClick={() => handleDelete(category)}
        >
          <svg
            xmlns="http://www.w3.org/2000/svg"
            className="h-6 w-6"
            fill="none"
            viewBox="0 0 24 24"
            stroke="currentColor"
          >
            <path
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              d="M6 18L18 6M6 6l12 12"
            />
          </svg>
        </button>
      </div>

      {isLoading && <h2>Cargando..</h2>}

      <div className="card-grid">
        {images.map((image) => (
          <GifItem
            key={image.id}
            {...image}
            //le mando todas las propiedades de imagen (url, title, id)
          />
        ))}
      </div>
    </>
  );
};
