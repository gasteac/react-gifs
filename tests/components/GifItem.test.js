import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('Evaluar GifItem', () => { 
    const title = 'SAITAMA';
    const url = 'https://one-punch.com/saitama.jpg'
    test('Debe hacer match con snapshot', () => { 

        const {container} = render(<GifItem title={title} url={url}/>); 
        expect(container).toMatchSnapshot();
    })
    test('debe mostrar la imagen con el URL y el ALT indicado', () => { 
        render(<GifItem title={title} url={url}/>)
        //screen.debug();
        // expect(screen.getByRole('img').src).toBe(url)
        // expect(screen.getByRole('img').alt).toBe(title)
        const {src, alt} = screen.getByRole('img')
        expect(src).toBe(url);
        expect(alt).toBe(title);
     })
     test('debe de mostrar el titulo en el componente', () => { 
        render(<GifItem title={title} url={url}/>)
        expect(screen.getByText(title)).toBeTruthy();
      })
 })