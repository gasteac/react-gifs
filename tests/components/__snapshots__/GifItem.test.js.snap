// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Evaluar GifItem Debe hacer match con snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="SAITAMA"
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
      SAITAMA
    </p>
  </div>
</div>
`;
